### PR TITLE
feat(js): LANZ-2248 Improved gh action to accept scope and breaking change exclamation mark

### PR DESCRIPTION
**Pull request by cangulomascarell**

### CODE REVIEW

File: ensure-conventional-commits.js [hunk 1]
```diff
@@ -72,6 +75,33 @@ if (!ok) {
 // --------------------- //
 // ----- FUNCTIONS ----- //
 // --------------------- //
+function validateCommitFormat(commitMsg, commitTypesAccepted) {
```
Comment (cangulomascarell) [inline ensure-conventional-commits.js:78]: commit examples:
* fix(js)!: commit type fix but with a breaking change
* fix(js): commit type fix with scope js
* feat: commit type feat
* break: commit type break
* docs: commit type docs
* refactor: commit type refactor

All of them have `:` 
format: preffix (scope/optional) : msg

File: ensure-conventional-commits.js [hunk 2]
```diff
@@ -72,6 +75,33 @@ if (!ok) {
...
+    if (!validCommitType) {
+      return false
+    }
+
+    const scopeRegex = /(?<type>^[a-z]+)(?<scope>\([a-z,\-]+\))?(?<breaking>!)?(?<colon>:{1})/
+    let matchResult = commitMsg.match(scopeRegex);
+    if (!matchResult) {
+      return false
+    }
+
```
Comment (cangulomascarell) [inline ensure-conventional-commits.js:93]: image

File: logging.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
 // https://stackoverflow.com/questions/9781218/how-to-change-node-jss-console-font-color
```
Comment (cangulomascarell) [inline logging.js:3]: fixed error in the logging. now it display yellos properly:
image

**Review by cangulomascarell — COMMENTED** [review]
here it is an execution example:
https://github.com/ohpen/platform-repository-for-testing/runs/5989683399?check_suite_focus=true

File: ensure-conventional-commits.js [hunk 4]
```diff
@@ -72,6 +75,33 @@ if (!ok) {
...
+    let { breaking } = matchResult.groups;
+    if (breaking == '!') {
+      logger.logWarning(`this commits is a breaking change`);
```
Comment (jstehlik-ohpen) [inline ensure-conventional-commits.js]: commits -> commit

**Review by jcajkovic — DISMISSED** [review]

File: ensure-conventional-commits.js (hunk 4 above)
Comment (cangulomascarell) [reply ensure-conventional-commits.js]: thanks!

**Pull request merged**